From d532d2a2309f635532b3d36adbfb576d8c14b85a Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Mon, 5 Sep 2022 16:20:38 +0200
Subject: [PATCH 09/21] daemon: Don't warn if a device gets unexported

Now that we can unregister devices from the bus, make sure that we don't
throw warnings doing that.
---
 src/up-daemon.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/up-daemon.c b/src/up-daemon.c
index 5bf9eaf..333f73b 100644
--- a/src/up-daemon.c
+++ b/src/up-daemon.c
@@ -970,13 +970,13 @@ up_daemon_device_added_cb (UpBackend *backend, UpDevice *device, UpDaemon *daemo
 
 	/* emit */
 	object_path = up_device_get_object_path (device);
-	g_debug ("emitting added: %s", object_path);
-
-	/* don't crash the session */
 	if (object_path == NULL) {
-		g_warning ("INTERNAL STATE CORRUPT (device-added): not sending NULL, device:%p", device);
+		g_debug ("Device %s was unregistered before it was on the bus",
+			 up_exported_device_get_native_path (UP_EXPORTED_DEVICE (device)));
 		return;
 	}
+
+	g_debug ("emitting added: %s", object_path);
 	up_daemon_update_warning_level (daemon);
 	up_exported_daemon_emit_device_added (UP_EXPORTED_DAEMON (daemon), object_path);
 }
-- 
2.39.1

